effect/jvm/src/test/scala/scalaz/effect/generator/GenIO.scala

```diff
@@ -0,0 +1,150 @@
+package scalaz.effect.generator
+
+import org.scalacheck._
+import scalaz.effect.{ ExitResult, IO }
+
+trait GenIO {
+
+  /**
+   * Given a generator for `A`, produces a generator for `IO[E, A]` using the `IO.point` constructor.
+   */
+  def genSyncSuccess[E, A: Arbitrary]: Gen[IO[E, A]] =
+    for {
+      a <- Arbitrary.arbitrary[A]
+    } yield IO.point[E, A](a)
+
+  /**
+   * Given a generator for `A`, produces a generator for `IO[E, A]` using the `IO.async` constructor.
+   */
+  def genAsyncSuccess[E, A: Arbitrary]: Gen[IO[E, A]] =
+    for {
+      a <- Arbitrary.arbitrary[A]
+    } yield IO.async[E, A](cb => cb(ExitResult.Completed(a)))
+
+  /**
+   * Randomly uses either `genSyncSuccess` or `genAsyncSuccess` with equal probability.
+   */
+  def genSuccess[E, A: Arbitrary]: Gen[IO[E, A]] = Gen.oneOf(genSyncSuccess[E, A], genAsyncSuccess[E, A])
+
+  /**
+   * Given a generator for `E`, produces a generator for `IO[E, A]` using the `IO.fail` constructor.
+   */
+  def genSyncFailure[E: Arbitrary, A]: Gen[IO[E, A]] =
+    for {
+      error <- Arbitrary.arbitrary[E]
+    } yield IO.fail(error)
+
+  /**
+   * Given a generator for `E`, produces a generator for `IO[E, A]` using the `IO.async` constructor.
+   */
+  def genAsyncFailure[E: Arbitrary, A]: Gen[IO[E, A]] =
+    for {
+      error <- Arbitrary.arbitrary[E]
+    } yield IO.async(cb => cb(ExitResult.Failed(error)))
+
+  /**
+   * Randomly uses either `genSyncFailure` or `genAsyncFailure` with equal probability.
+   */
+  def genFailure[E: Arbitrary, A]: Gen[IO[E, A]] = Gen.oneOf(genSyncFailure[E, A], genAsyncFailure[E, A])
+
+  /**
+   * Randomly uses either `genSuccess` or `genFailure` with equal probability.
+   */
+  def genIO[E: Arbitrary: Cogen, A: Arbitrary: Cogen]: Gen[IO[E, A]] =
+    Gen.oneOf(genSuccess[E, A], genFailure[E, A])
+
+  /**
+   * Given a generator for `IO[E, A]`, produces a sized generator for `IO[E, A]` which represents a transformation,
+   * by using some random combination of the methods `map`, `flatMap`, `leftMap`, and any other method that does not change
+   * the success/failure of the value, but may change the value itself.
+   */
+  def genLikeTrans[E: Arbitrary: Cogen, A: Arbitrary: Cogen](gen: Arbitrary[IO[E, A]]): Gen[IO[E, A]] = {
```

Comment:
I would never pass `Arbitrary` as a parameter, but rather, only pass `Gen`. `Arbitrary` exists only as a type class that allows you to summon a `Gen` for some type `A`. Although you can pass type class instances in Scala, if there is a way to avoid it, I generally find the code cleaner.